### PR TITLE
Added median for array

### PR DESCRIPTION
**Pull request by Narendra-Pandey**
## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 5.0.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+, or use `@available` if not.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [ ] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

**Comment by omaralbeik** [comment]
@Narendra-Pandey Consider using [SwiftLint](https://github.com/realm/SwiftLint) to lint your code, it is already added to SwifterSwift, all you have to do is install it on your computer

File: Sources/SwifterSwift/SwiftStdlib/ArrayExtensions.swift [hunk 1]
```diff
@@ -19,7 +19,18 @@ public extension Array {
         insert(newElement, at: 0)
     }
 
-    /// SwifterSwift: Safely swap values at given index positions.
+  /// SwifterSwift: Insert an element in the middle of the collection.
```
Comment (LucianoPAlmeida) [inline Sources/SwifterSwift/SwiftStdlib/ArrayExtensions.swift]: Nit: Identation

File: Sources/SwifterSwift/SwiftStdlib/ArrayExtensions.swift [hunk 2]
```diff
@@ -19,7 +19,18 @@ public extension Array {
...
+  ///
+  ///       [1, 2, 3, 4].insertInMiddle(5) -> [1, 2, 5, 3 , 4]
+  ///       [1, 2, 3, 4, 5].insertInMiddle(6) -> [1, 2, 6, 3 , 4, 5]
+  ///
+  /// - Parameter newElement: element to insert.
+  mutating func insertInMiddle(_ newElement: Element) {
+      let middleIndex = index(startIndex, offsetBy: distance(from: startIndex, to: endIndex) / 2)
```
Comment (LucianoPAlmeida) [inline Sources/SwifterSwift/SwiftStdlib/ArrayExtensions.swift]:
```suggestion
      let middleIndex = index(startIndex, offsetBy: count/ 2)
```
Distance from `start` to `end` is the same as `count` :)

File: Tests/SwiftStdlibTests/ArrayExtensionsTests.swift [hunk 3]
```diff
@@ -15,7 +15,21 @@ final class ArrayExtensionsTests: XCTestCase {
         arr.prepend(1)
         XCTAssertEqual(arr, [1, 2, 3, 4, 5])
     }
-
+  
+    func testInsertInMiddle() {
```
Comment (LucianoPAlmeida) [inline Tests/SwiftStdlibTests/ArrayExtensionsTests.swift]: Nit: Identation

File: Tests/SwiftStdlibTests/ArrayExtensionsTests.swift [hunk 4]
```diff
@@ -15,7 +15,21 @@ final class ArrayExtensionsTests: XCTestCase {
...
+      var array = [1, 2, 3, 4]
+      array.insertInMiddle(5)
+      XCTAssertEqual(array, [1, 2, 5, 3, 4])
+      
+      array = [1, 2, 3, 4, 5]
+      array.insertInMiddle(6)
+      XCTAssertEqual(array, [1, 2, 6, 3, 4, 5])
+
```
Comment (LucianoPAlmeida) [inline Tests/SwiftStdlibTests/ArrayExtensionsTests.swift]: Can you add a test case for a single-element array e.g. `array = [1]`?

**Comment by LucianoPAlmeida** [comment]
Closing due inactivity, @Narendra-Pandey feel free to reopen anytime :))

**Pull request closed without merging**